backends/bmv2/simple_switch/midend.cpp

```diff
@@ -125,6 +125,7 @@ SimpleSwitchMidEnd::SimpleSwitchMidEnd(CompilerOptions& options, std::ostream* o
             isv1 ? new P4::RemoveUnusedActionParameters(&refMap) : nullptr,
             new P4::TypeChecking(&refMap, &typeMap),
             options.loopsUnrolling ? new P4::ParsersUnroll(true, &refMap, &typeMap) : nullptr,
+            new P4::TypeChecking(&refMap, &typeMap, true),
```

Comment:
This is not late enough, there are many passes run in the backend that could potentially modify this.
Look for example at simpleSwitch.cpp.